CIP-0000.md

```diff
@@ -0,0 +1,198 @@
+
+Because the CIPs are maintained as text files in a versioned repository, their revision history is the historical record of the feature proposal.
+
+Participation in the Celo project and the CIP process is informed by the Celo [Community Tenets](https://celo.org/community) and subject to the  [Code of Conduct](https://celo.org/code-of-conduct).
+
+# CIP Types
+There are three types of CIP:
+
+* A **Standards Track CIP** describes any change that affects most or all Celo implementations, such as a change to the network protocol, a change in block or transaction validity rules, proposed application standards/conventions, or any change or addition that affects the interoperability of applications using Celo. Furthermore Standard CIPs can be broken down into the following categories. Standards Track CIPs consist of two parts, a design document and implementation.
+	* **Core** - improvements requiring a consensus fork, as well as changes that are not necessarily consensus critical but may be relevant to  “core dev” discussions, and the validator/node strategy.
+	* **Governance** - improvements to the governance proposal to the Celo platform including parameter changes. It should explain the reason for the change, enumerate the smart contract methods that will be called with parameters, and describe the impact and any risks. 
+	* **Networking** - improvements to networking protocol specifications such as devp2p, and proposed improvements like whisper and swarm.
+	* **Interface** - improvements around client [API/RPC] specifications and standards, and also certain language-level standards like method names and contract ABIs.
+	* **CRC** - (Celo [Request for Comments](https://en.wikipedia.org/wiki/Request_for_Comments)) application-level standards and conventions, including contract standards such as token standards, name registries, URI schemes, library/package formats, and wallet formats.
+* A **Meta CIP** describes a process surrounding Celo or proposes a change to (or an event in) a process. Process CIPs are like Standards Track CIPs but apply to areas other than the Celo protocol itself. They may propose an implementation, but not to Celo’s codebase; they often require community consensus; unlike Informational CIPs, they are more than recommendations, and users are typically not free to ignore them. Examples include code of conduct updates, procedures, guidelines, changes to the decision-making process, and changes to the tools or environment used in Celo development. Any Meta CIP is also considered a Process CIP.
+* An **Informational CIP** describes a Celo design issue, or provides general guidelines or information to the Celo community, but does not propose a new feature. Informational CIPs do not necessarily represent Celo community consensus or a recommendation, so users and implementers are free to ignore Informational CIPs or follow their advice.
+
+It is highly recommended that a single CIP contain a single key proposal or new idea.
```

Comment:
How do you see this wrt smart contract upgrades? The tooling currently supports upgrading all contracts at once rather than individually to ensure compatibility and a clear audit trail

Reply:
It's a good question, I guess it goes back to whether the smart contract upgrade is a parameter change that's part of defining the Celo standard or if it's something more related to maintenance and refactoring. Perhaps smart contracts need to be part of the CGP instead.

Reply:
IMO, the stability protocol, for example, is part of the Celo standard, and changes to the stability protocol (e.g. a different formula for calculating bucket sizes) fit the "core" category.

It's tricky, because adopting such a change will ultimately result in a governance proposal. In our existing process, changes like this would first go through as a CIP, after which they would be merged into master, and then later as a CGP to upgrade all contracts to the latest on `master` (which would effectively bundle a number of CIPs, as well as changes to contracts too small to necessitate a CIP).

Curious to get your thoughts on all of this, and how you were thinking core contract changes fit into the proposed process.